src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -83,5 +118,194 @@ public void AssemblyLoading()
 
             Assert.Equal(expected, actual);
         }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void AssemblyLoading_AssemblyLocationNotAdded()
+        {
+            var loader = new DefaultAnalyzerAssemblyLoader();
+            loader.AddDependencyLocation(_testFixture.Gamma.Path);
+            loader.AddDependencyLocation(_testFixture.Delta1.Path);
+            Assert.Throws<FileNotFoundException>(() => loader.LoadFromPath(_testFixture.Beta.Path));
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void AssemblyLoading_DependencyLocationNotAdded()
+        {
+            StringBuilder sb = new StringBuilder();
+            var loader = new DefaultAnalyzerAssemblyLoader();
+            loader.AddDependencyLocation(_testFixture.Gamma.Path);
+            loader.AddDependencyLocation(_testFixture.Beta.Path);
+            Assembly beta = loader.LoadFromPath(_testFixture.Beta.Path);
+
+            var b = beta.CreateInstance("Beta.B")!;
+            var writeMethod = b.GetType().GetMethod("Write")!;
+            var exception = Assert.Throws<TargetInvocationException>(
+                () => writeMethod.Invoke(b, new object[] { sb, "Test B" }));
+            Assert.IsAssignableFrom<FileNotFoundException>(exception.InnerException);
+
+            var actual = sb.ToString();
+            Assert.Equal(@"", actual);
+        }
+
+        [Fact]
+        public void AssemblyLoading_MultipleVersions()
```

Comment:
This is the type of test that I think would benefit from inspecting the `AssemblyLoadContext` directly. Essentially after the test completes looking at the loaded `Assembly` objects and verifying via items like `Location` that they match up with the expected inputs.

Reply:
I tried out a test accessor for the ALC collection in this test.